src/network/server.rs

```diff
@@ -1008,8 +1009,67 @@ impl HubService for MyHubService {
                 events
             })
             .collect();
-        let combined_events: Vec<HubEvent> =
+        let mut combined_events: Vec<HubEvent> =
```

Comment:
One thing to note here is that you're doing the filtering after you pull the events out of the db. This means that the user might receive empty responses with a pointer to a next page token to look at because there are no valid events in this page but there are still more events to look at. I think we should do this filtering while pulling events out of the db so we always return as full of a response as we can. Here's an example from another store: https://github.com/farcasterxyz/snapchain/blob/81c7e8ecbb6f110cacdf72a8d52d4bae3144dc9d/src/storage/store/account/onchain_event_store.rs#L301

Reply:
There might be some complications with this because we don't store event timestamp and block number directly in the db. If that's a blocker, we should just start storing those values in the db.

Reply:
Yeah, I think we should get this change in along with this feature. I don't necessarily see this as an optimization-- more of a UX improvement for users calling the rpc. They may have to make a lot of rpc calls to get a very small amount of data unless we try to fill pages.